### PR TITLE
More fixes to get yklua working.

### PR DESCRIPTION
**Pull request by ptersilie**
See commits for more details.

### CODE REVIEW

File: ykllvmwrap/src/jitmodbuilder.cc [hunk 1]
```diff
@@ -74,7 +74,9 @@ struct BlockResumePoint {
   //
   // If we want to resume at the `load`, then we store the `call`.
   Instruction *ResumeAfterInstr;
-  // The index of the instruction immediately *before* where we should resume.
+  // The index of the last call instruction. This is the instruction immediately
+  // *before* the instruction we want to resume from after returning from a
+  // basic block.
```
Comment (vext01) [inline ykllvmwrap/src/jitmodbuilder.cc:79]: s/basic block/call/ ?

File: ykllvmwrap/src/jitmodbuilder.cc [hunk 2]
```diff
@@ -176,10 +178,11 @@ struct MappableFrame {
     }
   }
 
-  // Return the instruction immediately before the instruction to resume at.
+  // Return the last encountered call in the current basic block. This is the call
+  // we need to resume from after processing the callee.
```
Comment (vext01) [inline ykllvmwrap/src/jitmodbuilder.cc:182]: s/resume from/resume after/ ?

File: ykfr/src/lib.rs [hunk 3]
```diff
@@ -306,7 +306,10 @@ impl FrameReconstructor {
                     }
                     SMLocation::Direct(reg, off, _) => {
                         if i == 0 {
-                            // skip first frame
+                            // FIXME: In contrast to indirect locations, we don't need to copy
+                            // values over to the bottom frame. In fact, doing so results in
+                            // segfaults. It is not clear to me as to why that is. Requires
+                            // investigation.
```
Comment (vext01) [inline ykfr/src/lib.rs:312]: Curious...

Comment (vext01) [reply ykfr/src/lib.rs:312]: I think you should investigate this after this PR is merged while it is still fresh in your mind.

File: ykfr/src/lib.rs [hunk 4]
```diff
@@ -324,14 +327,23 @@ impl FrameReconstructor {
...
+                            // any changes directly to it from here.
+                            unsafe { btmframeaddr.offset(isize::try_from(*off).unwrap()) }
+                        } else {
+                            unsafe { rbp.offset(isize::try_from(*off).unwrap()) }
+                        };
                         debug_assert!(*off < i32::try_from(rec.size).unwrap());
+                        // FIXME: The minimum size reported by the stackmap is 1 which represents 1
+                        // byte. LLVM IR allows for smaller sizes, e.g. `i1` representing a single
+                        // bit. It is currently unclear how that affects this code, so I'm leaving
+                        // this comment here so we don't forget.
```
Comment (vext01) [inline ykfr/src/lib.rs:344]: I think LLVM would use a whole byte to store an `i1`.

In theory you could pack a load of `i1`s into a byte, but then it raises questions about what it means to take the address of the bits...

**Comment by vext01** [comment]
LGTM modulo a few bits.

File: ykllvmwrap/src/jitmodbuilder.cc (hunk 1 above)
Comment (ptersilie) [reply ykllvmwrap/src/jitmodbuilder.cc]: [e5ad88a](https://github.com/ykjit/yk/pull/637/commits/e5ad88ac717eda283bea283e39da113f102dff07)

File: ykllvmwrap/src/jitmodbuilder.cc (hunk 2 above)
Comment (ptersilie) [reply ykllvmwrap/src/jitmodbuilder.cc]: [e5ad88a](https://github.com/ykjit/yk/pull/637/commits/e5ad88ac717eda283bea283e39da113f102dff07)

File: ykfr/src/lib.rs (hunk 3 above)
Comment (ptersilie) [reply ykfr/src/lib.rs]: I updated this with more informationin [acb97d1](https://github.com/ykjit/yk/pull/637/commits/acb97d1e094316d83e44a89fdea608f5a078a186). There's still some remaining questions, so I left another fixme.

File: ykfr/src/lib.rs (hunk 4 above)
Comment (ptersilie) [reply ykfr/src/lib.rs:346]: I think we should leave the comment anyway.

**Comment by ptersilie** [comment]
Think this is ready to go after squashing.

**Comment by vext01** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**